contracts/Tokens/VAI/VAIController.sol

```diff
@@ -387,6 +394,34 @@ contract VAIController is VAIControllerStorageG2, VAIControllerErrorReporter, Ex
         return uint(Error.NO_ERROR);
     }
 
+    /**
+     * @notice Set the prime token contract address
+     * @param prime_ The new address of the prime token contract
+     */
+    function _setPrimeToken(address prime_) external onlyAdmin {
+        emit NewPrime(prime, prime_);
+        prime = prime_;
+
+    }
+
+    /**
+     * @notice Toggle mint only for prime holder
+     * @param _mintEnabledOnlyForPrimeHolder set to true to allow minting only for prime holders
+     * @return uint 0=success, otherwise a failure (see ErrorReporter.sol for details)
+     */
+    function toggleOnlyPrimeHolderMint(bool _mintEnabledOnlyForPrimeHolder) external returns (uint) {
```

Comment:
I would call it `setOnlyPrimeHolderMint` if we are going to accept a parameter, or I would remove the parameter if we are going to keep this name (and I would do this internally: `mintEnabledOnlyForPrimeHolder = !mintEnabledOnlyForPrimeHolder`)